### DISCUSSION TITLE
CTkMessagebox

### DISCUSSION THREAD

**Discussion by Akascape** [TomSchimansky/CustomTkinter#1279] [Showcase]
I made a fully customizable messagebox for customtkinter!

# [CTkMessagebox](https://github.com/Akascape/CTkMessagebox)
![Screenshot](https://user-images.githubusercontent.com/89206401/221258593-75058878-b598-40c3-828a-1d44a6cefb73.jpg)

### Hope it will help!

**Comment by VollGerne** [comment]
Very nice, thanks....
I will use it in one of my projects.

**Comment by CoderRichit** [comment]
Awesome Bro. Thank you very much !!!!
😀

**Comment by avalon60** [comment]
Akash, this looks nice! 

A couple of points though. In Pyrhon3, the standard is that module names should be in lower case and words often (although for some reason discouraged) separated by underscore.  Please see [PEP-8](https://pymbook.readthedocs.io/en/latest/pep8.html#package-and-module-names). 
So you might expect to see something like:

`import ctk_messagebox`

OR

`from ctk_messagebox import CTkMessagebox`

Also I tried this with CustomTkinter version 4 and it doesn't work, presumably the __apply_appearance_mode_ method, is specific to CustomTkinter 5? 

I got the exception: _AttributeError: 'CTkMessagebox' object has no attribute '_apply_appearance_mode'_

Thanks,

Clive

**Comment by Akascape** [comment]
@avalon60

> Also I tried this with CustomTkinter version 4 and it doesn't work, presumably the _apply_appearance_mode method, is specific to CustomTkinter 5?

Obviously, it is not made for customtkinter 4. Only works with latest version.

**Reply by avalon60** [reply]
So it will stop working in version 5, when version 6 comes out? ;)

**Reply by Akascape** [reply]
@avalon60 I created this package after ctk 5. That is why there is no version for ctk 4.

Will make it compatible whenever new updates are released for customtkinter.

Moreover, I have ported most of my apps and packages to ctk 5. And everyone should do it so that they can use the new widgets and features.

**Reply by avalon60** [reply]
Migrating to the latest and greatest, sometimes takes time, especially for larger projects.

**Comment by RunsOnRum** [comment]
@Akascape, would it be possible to make the messages popup center to the window calling the event by default? Currently, if the application is on another display, the popup will be displayed on the main display (win10).

**Reply by Akascape** [reply]
@RunsOnRum 
Just pass the `master=root` in the messagebox args.

**Reply by RunsOnRum** [reply]
My apologies. If I only had taken the time to read the instructions ;-) Thank you for your good work!
